clippy_lints/src/if_let_mutex.rs

```diff
@@ -45,6 +46,10 @@ declare_lint_pass!(IfLetMutex => [IF_LET_MUTEX]);
 
 impl<'tcx> LateLintPass<'tcx> for IfLetMutex {
     fn check_expr(&mut self, cx: &LateContext<'tcx>, expr: &'tcx Expr<'tcx>) {
+        if cx.tcx.sess.edition() >= Edition2024 {
```

Comment:
Can you update the lint description to mention that this lint is effectively disabled starting in edition 2024 as `if let` scoping was reworked such that this is no longer an issue?